src/Metrics/Transports/OpenTelemetryMetricTransport.php

```diff
@@ -0,0 +1,102 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\OpenTelemetry\Metrics\Transports;
+
+use OpenTelemetry\API\Metrics\MeterInterface;
+use OpenTelemetry\API\Metrics\ObserverInterface;
+use OpenTelemetry\Context\ContextInterface;
+use Shopware\Core\Framework\Telemetry\Metrics\Exception\MetricNotSupportedException;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Counter;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Gauge;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\Histogram;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\MetricInterface;
+use Shopware\Core\Framework\Telemetry\Metrics\Metric\UpDownCounter;
+use Shopware\Core\Framework\Telemetry\Metrics\MetricTransportInterface;
+
+/**
+ * @phpstan-type Attributes iterable<non-empty-string, string|bool|float|int>
+ */
+class OpenTelemetryMetricTransport implements MetricTransportInterface
+{
+    public function __construct(
+        private string $namespace,
+        private MeterInterface $meter,
+    ) {
+    }
+
+    public function emit(MetricInterface $metric): void
+    {
+        $attributes = [];
+        $context = null;
+
+        match (true) {
+            $metric instanceof UpDownCounter => $this->handleUpDownCounter($metric, $attributes, $context),
+            $metric instanceof Counter => $this->handleCounter($metric, $attributes, $context),
+            $metric instanceof Histogram => $this->handleHistogram($metric, $attributes, $context),
+            $metric instanceof Gauge => $this->handleGauge($metric, $attributes, $context),
+            default => throw new MetricNotSupportedException(
```

Comment:
based on discussion, this was updated to be:

```suggestion
            default => \Shopware\Core\Framework\Telemetry\Metrics\Exception\MetricsException::metricNotSupported($metric, $this)
```

Reply:
Changed exception creation.

Regarding cs-fixer and php-stan sure, let's do it as a separate change.